File: android/RecyclerView-master/RecyclerView-master/app/src/main/java/com/dicoding/picodiploma/recyclerviewsubmission/MainActivity.java
package com.dicoding.picodiploma.recyclerviewsubmission;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.widget.RelativeLayout;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    private RecyclerView rvPlayers;
    private ArrayList<Player> list = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        rvPlayers = findViewById(R.id.rv_players);
        rvPlayers.setHasFixedSize(true);

        list.addAll(PlayerData.getListData());
        showRecyclerList();
    }

    private void showRecyclerList(){
        rvPlayers.setLayoutManager(new LinearLayoutManager(this));
        ListAdapter listAdapter = new ListAdapter(list);
        rvPlayers.setAdapter(listAdapter);
    }


    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.menu_main,menu);
        return true;
    }

    public void Pindah(MenuItem item) {
        Intent intent = new Intent(MainActivity.this, AboutActivity.class);
        startActivity(intent);
    }

}

File: android/20220224190841ee30b7481bce4919a6af41f6a270fae9/GithubUserApp2/app/src/main/java/com/frhnfath/githubuserapp/activity/MainActivity.kt
package com.frhnfath.githubuserapp.activity

import android.app.SearchManager
import android.content.Context
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Menu
import android.view.View
import androidx.appcompat.widget.SearchView
import android.widget.Toast
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import com.frhnfath.githubuserapp.R
import com.frhnfath.githubuserapp.adapter.UserAdapter
import com.frhnfath.githubuserapp.response.UserResponse
import com.frhnfath.githubuserapp.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private val list = ArrayList<UserResponse>()
    private lateinit var binding: ActivityMainBinding
    private lateinit var userAdapter: UserAdapter
    private lateinit var viewModel: MainViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        userAdapter = UserAdapter()
        userAdapter.notifyDataSetChanged()
        userAdapter.setOnItemClickCallback(object : UserAdapter.OnItemClickCallback {
            override fun onItemClicked(data: UserResponse) {
                Intent(this@MainActivity, DetailActivity::class.java).also {
                    it.putExtra(DetailActivity.EXTRA_USERNAME, data.login)
                    it.putExtra(DetailActivity.EXTRA_NAME, data.name)
                    it.putExtra(DetailActivity.EXTRA_COMPANY, data.company)
                    showSelectedUser(data)
                    startActivity(it)
                }
            }

        })

        showRecyclerList()
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        val inflater = menuInflater
        inflater.inflate(R.menu.options_menu, menu)

        val searchView = menu.findItem(R.id.search).actionView as SearchView
        val searchManager = getSystemService(Context.SEARCH_SERVICE) as SearchManager

        searchView.setSearchableInfo(searchManager.getSearchableInfo(componentName))
        searchView.queryHint = resources.getString(R.string.search_hint)
        searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
            override fun onQueryTextSubmit(query: String): Boolean {
                showProgressBar(true)
                viewModel.searchUser(query)
                return true
            }

            override fun onQueryTextChange(newText: String): Boolean {
                list.clear()
                return false
            }
        })
        return true
    }

    private fun showProgressBar(b: Boolean) {
        if (b)  {
            binding.progressBarMain.visibility = View.VISIBLE
        }
        else {
            binding.progressBarMain.visibility = View.GONE
        }
    }


    // create function to show recycler list
    private fun showRecyclerList() {
        viewModel = ViewModelProvider(this, ViewModelProvider.NewInstanceFactory()).get(
            MainViewModel::class.java)
        binding.apply {
            rvUsers.layoutManager = LinearLayoutManager(this@MainActivity)
            rvUsers.setHasFixedSize(true)
            rvUsers.adapter = userAdapter
        }

        viewModel.getSearch().observe(this) { query ->
            if (query != null) {
                userAdapter.setList(query)
                showProgressBar(false)
            }
        }
    }


    fun showSelectedUser(user: UserResponse) {
        // create Toast to enable flash information to user
        Toast.makeText(this, "You choose " + user.login, Toast.LENGTH_SHORT).show()
    }
}